incubator/bootstrap_cli/__main__.py

```diff
@@ -224,206 +201,362 @@ def __init__(self, message: str):
+        self.client: CogniteClient = None
+        self.cdf_project = None
 
         # TODO debug
         # print(f"self.config= {self.config}")
 
+        # TODO: support 'logger' section optional, provide default config for logger with console only
+        #
+        # Logger initialisation
+        #
         # make sure the optional folders in logger.file.path exists
         # to avoid: FileNotFoundError: [Errno 2] No such file or directory: '/github/workspace/logs/test-deploy.log'
-
         if self.config.logger.file:
             (Path.cwd() / self.config.logger.file.path).parent.mkdir(parents=True, exist_ok=True)
-
         self.config.logger.setup_logging()
-
         _logger.info("Starting CDF Bootstrap configuration")
 
-        self.client: CogniteClient = self.config.cognite.get_cognite_client(  # noqa
-            client_name="inso-bootstrap-cli", token_custom_args=self.config.token_custom_args
-        )
+        # debug new features
+        if getattr(self, "bootstrap_config", False):
+            # TODO: not available for 'delete' but there must be aa smarter solution
+            _logger.info(
+                "Features from yaml-config or defaults (can be overridden by cli-parameters!): "
+                f"{self.bootstrap_config.features=}"
+            )
 
-        _logger.info("Successful connection to CDF client")
+        #
+        # Cognite initialisation (optional for 'diagram')
+        #
+        if self.config.cognite:
+            self.client: CogniteClient = self.config.cognite.get_cognite_client(  # noqa
+                client_name="inso-bootstrap-cli", token_custom_args=self.config.token_custom_args
+            )
+            self.cdf_project = self.client.config.project
+            _logger.info("Successful connection to CDF client")
 
     @staticmethod
     def acl_template(actions, scope):
         return {"actions": actions, "scope": scope}
 
     @staticmethod
-    def get_allprojects_name_template(group_ns=None):
-        return f"{group_ns}:{BootstrapCore.AGGREGATED_GROUP_NAME}" if group_ns else BootstrapCore.AGGREGATED_GROUP_NAME
+    def get_allprojects_name_template(ns_name=None):
+        return f"{ns_name}:{BootstrapCore.AGGREGATED_LEVEL_NAME}" if ns_name else BootstrapCore.AGGREGATED_LEVEL_NAME
 
     @staticmethod
     def get_dataset_name_template():
-        return "{group_core}:dataset"
+        return "{node_name}" + BootstrapCore.DATASET_SUFFIX
 
     @staticmethod
     def get_raw_dbs_name_template():
-        return "{group_core}:rawdb{raw_suffix}"
+        return "{node_name}" + BootstrapCore.RAW_SUFFIX + "{raw_variant}"
 
     @staticmethod
     def get_timestamp():
         return datetime.now().strftime("%Y-%m-%d %H:%M:%S")
 
+    def validate_config_length_limits(self):
+        """
+        Validate features in config
+        """
+
+        #
+        # CHECK 1 (availability)
+        #
+        assert self.AGGREGATED_LEVEL_NAME, (
```

Comment:
should this not be checked if running in optimized flows?:
https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement